src/e3/anod/deps.py

```diff
@@ -79,8 +79,24 @@ def __init__(
         self.host = host
         self.target = target
         self.build = build
-        self.qualifier = qualifier
         self.local_name = local_name if local_name is not None else name
+
+        self.qualifier: str | None
+        if isinstance(qualifier, dict):
+            for q in set(qualifier.keys()):
+                v = qualifier[q]
```

Comment:
for q, v in qualifier.items()

Reply:
I cannot do that because I am changing the size of the dictionary in the loop (poping an element) and python doesn't like that.